library/Ivoz/Api/Doctrine/Orm/Extension/UnpaginatedResultGeneratorExtension.php

```diff
@@ -0,0 +1,194 @@
+    public function getResult(
+        QueryBuilder $queryBuilder,
+        string $resourceClass = null,
+        string $operationName = null,
+        array $context = []
+    ) {
+        ini_set('max_execution_time', 0);
+
+        $sqlLogger = $this
+            ->entityManager
+            ->getConnection()
+            ->getConfiguration()->getSQLLogger();
+
+        $this
+            ->entityManager
+            ->getConnection()
+            ->getConfiguration()->setSQLLogger(null);
+
+
+        $queryBuilder
+            ->setMaxResults(self::BATCH_SIZE);
+
+        $currentPage = 1;
+        $continue = true;
+
+        while ($continue) {
+            $queryBuilder
+                ->setFirstResult(
+                    ($currentPage -1) * self::BATCH_SIZE
+                );
+
+            $query = $queryBuilder->getQuery();
+            $results = $query->getResult();
+            $continue = count($results) === self::BATCH_SIZE;
+            $currentPage++;
+
+            foreach ($results as $entity) {
+                yield $entity;
+                $this->entityManager->detach($entity);
+            }
+            $this->entityManager->clear();
+        }
+
+        $maxMemoryUsage = $this->getMemoryLimit();
+        $currentMemoryUsage = memory_get_usage(false);
+
+        if ($currentMemoryUsage/$maxMemoryUsage > 0.8) {
+            $this->logger->warning(
+                'Unpaginated request memory usage is over 80%: '
+                . round(memory_get_usage(false) / 1024/ 1024, 1) . 'MB'
+            );
+        }
+
+        $this
+            ->entityManager
+            ->getConnection()
+            ->getConfiguration()->setSQLLogger($sqlLogger);
+    }
+
+    private function isPaginationEnabled(
+        Request $request,
+        ResourceMetadata $resourceMetadata,
+        string $operationName = null
+    ): bool {
+        $enabled = $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_enabled', $this->enabled, true);
+        $clientEnabled = $resourceMetadata->getCollectionOperationAttribute($operationName, 'pagination_client_enabled', $this->clientEnabled, true);
+
+        if ($clientEnabled) {
+            $enabled = filter_var(
+                $this->getPaginationParameter($request, $this->enabledParameterName, $enabled),
+                FILTER_VALIDATE_BOOLEAN
+            );
+        }
+
+        return $enabled;
+    }
+
+    private function getPaginationParameter(
+        Request $request,
+        string $parameterName,
+        $default = null
+    ) {
+        if (null !== $paginationAttribute = $request->attributes->get('_api_pagination')) {
+            return array_key_exists($parameterName, $paginationAttribute) ? $paginationAttribute[$parameterName] : $default;
+        }
+
+        return $request->query->get($parameterName, $default);
+    }
+
+    /**
+     * @param $matches
+     * @return float|int|string
+     */
+    private function getMemoryLimit()
```

Comment:
You're now officially a C programmer. I wouldn't bother too much on memory usage over 80%. If the configured threshold is reached PHP will stop and an error will arise. Otherwise, everything is fine to me.

Reply:
xD

This made more sense before `web/rest: iterate during data serialization on unpaginated pages` commit. I'll remove it